src/NuGet.Clients/NuGet.Console/Utils/CommandUIUtilities.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.Shell.Interop;
+using NuGet.VisualStudio;
+using NuGetConsole.Implementation;
+
+namespace NuGetConsole
+{
+    public static class CommandUiUtilities
+    {
+        private static readonly Lazy<IVsInvalidateCachedCommandState> CommandStateCacheInvalidator;
+
+        static CommandUiUtilities()
+        {
+            CommandStateCacheInvalidator = new Lazy<IVsInvalidateCachedCommandState>(
```

Comment:
What is this API. 
I could not find any references to it publicly or in index. 
I can tell it's comming from interop, but I can't understand what makes it special.

Reply:
I'd imagine this is the: 

> finally, NuGet tells VS to invalidate the "Default project" combobox, which causes VS to requery the (new) cached value at VS's earliest convenience.

but I was just curious how you learned about it.

Reply:
That is exactly what it does.

Ryan Molden suggested I use it.

Documentation is lagging behind API availability.